pypeerassets/__main__.py

```diff
@@ -166,27 +168,31 @@ def deck_transfer(provider: Provider, deck: Deck,
 def find_card_transfers(provider: Provider, deck: Deck) -> Generator:
     '''find all <deck> card transfers'''
 
-    card_transfers = []
+    card_transfers = cast(Iterable, [])
 
     if isinstance(provider, RpcNode):
+        if deck.id is None:
+            raise Exception("deck.id required to listtransactions")
         batch_data = [('getrawtransaction', [i["txid"], 1] ) for i in provider.listtransactions(deck.id)]
         result = provider.batch(batch_data)
         if result is not None:
             card_transfers = [i['result'] for i in result if result]
     else:
+        if deck.p2th_address is None:
+            raise Exception("deck.p2th_address required to listtransactions")
         if provider.listtransactions(deck.p2th_address):
             card_transfers = (provider.getrawtransaction(i, 1) for i in
                               provider.listtransactions(deck.p2th_address))
         else:
             raise EmptyP2THDirectory({'error': 'No cards found on this deck.'})
 
-    def card_parser(args) -> list:
+    def card_parser(args: Tuple[Provider, Deck, dict]) -> list:
         '''this function wraps all the card transfer parsing'''
 
         provider = args[0]
         deck = args[1]
         raw_tx = args[2]
-        cards = []
+        cards = cast(list, [])
```

Comment:
What does this do? `cast`?

Reply:
It's this fella https://docs.python.org/3.5/library/typing.html#typing.cast

It just gives the static type checker a hint about the variable's type if it can't otherwise deduce it.

I think more recent python3.6 you can write something like:
```
cards: list=[]
```
But that syntax isn't available in python3.5? At least I think that's what mypy was telling me.

Reply:
Ok I get it.